lib/email/receiver.rb

```diff
@@ -109,7 +109,7 @@ def create_incoming_email
     end
 
     def process_internal
-      raise BouncedEmailError  if is_bounce?
+      raise BouncedEmailError if is_bounce? && !SiteSetting.enable_whispers?
```

Comment:
All those special cases for `is_bounced?` worry me. Wouldn't it be possible to create the whisper the first time we know it's a bounce? `is_bounced?` tries to lookup an `EmailLog` entry which should have all the information needed (`user_id` and `post_id`). Am I missing something?

Reply:
As per [this IF condition](https://github.com/discourse/discourse/blob/c3bc11b6b9185b6373fde1e02623686ebde4feee/lib/email/receiver.rb#L203) there is a chance to miss bounce_key & email_log. For now I will try to reduce special cases.

Yes, if it is always possible to find `EmailLog` then it's better to create whisper in first time.